# Pwskills

## Data Science Master

### Statistics Assignment

## Q1
Q1. What are the three measures of central tendency?

    The three measures of central tendency are:

     Mean: The mean is calculated by summing up all the values in a dataset and dividing the sum by the total number of values. It represents the average value of the data.

      Median: The median is the middle value in a dataset when the values are arranged in ascending or descending order. If there is an even number of values, the median is the average of the two middle values. The median represents the middle value that divides the dataset into two equal halves.

      Mode: The mode is the value or values that appear most frequently in a dataset. A dataset can have no mode (when no value is repeated), one mode (unimodal), two modes (bimodal), or more than two modes (multimodal). The mode represents the most common value(s) in the dataset.

    These measures provide different insights into the central tendency of a dataset and help summarize its characteristics.

## Q2

Q2. What is the difference between the mean, median, and mode? How are they used to measure the
central tendency of a dataset?


    The mean, median, and mode are measures of central tendency used to describe the center or typical value of a dataset. Here's a breakdown of their differences and how they are used:

     Mean: The mean is calculated by summing up all the values in a dataset and dividing the sum by the total number of values. It provides an average value that takes into account all the data points. The mean is sensitive to extreme values or outliers since it considers the magnitude of each value. It is commonly used when the data is approximately normally distributed or symmetrical.

     Median: The median is the middle value in a dataset when the values are arranged in ascending or descending order. If there is an even number of values, the median is the average of the two middle values. The median is not influenced by extreme values or outliers, as it focuses on the position of the values. It is useful when dealing with skewed distributions or when the presence of outliers may significantly affect the mean.

     Mode: The mode is the value or values that appear most frequently in a dataset. It represents the highest peak(s) in the distribution. The mode is useful for identifying the most common or popular value(s) in the dataset. It is not affected by outliers and can be applied to both numerical and categorical data.

     These measures are used to understand the central tendency of a dataset by providing a summary of the typical value or values. The choice of measure depends on the nature of the data and the research question. In cases where the data is normally distributed and without outliers, the mean may be the most appropriate measure. However, if the data is skewed or contains outliers, the median or mode may provide a more robust representation of the central tendency.

## Q3
Q3. Measure the three measures of central tendency for the given height data:
[178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]

    To measure the three measures of central tendency (mean, median, and mode) for the given height data [178, 177, 176, 177, 178.2, 178, 175, 179, 180, 175, 178.9, 176.2, 177, 172.5, 178, 176.5], we can calculate them as follows:

Mean:

To find the mean, we sum up all the values and divide the sum by the total number of values.


In [1]:
Mean = (178 + 177 + 176 + 177 + 178.2 + 178 + 175 + 179 + 180 + 175 + 178.9 + 176.2 + 177 + 172.5 + 178 + 176.5) / 16

In [2]:
print(Mean)

177.01875


Median:
    
To find the median, we first arrange the values in ascending order:

[172.5, 175, 175, 176, 176.2, 176.5, 177, 177, 178, 178, 178, 178.2, 178.9, 179, 180]
Since there are 16 values, the median is the average of the 8th and 9th values:


In [5]:
Median = (177 + 178) / 2

In [6]:
print(Median)

177.5


Mode:
    
To find the mode, we identify the value(s) that appear most frequently in the dataset.
In this case, the mode is 178, as it appears three times, which is more than any other value.

So, the measures of central tendency for the given height data are:

Mean ≈ 176.68125

Median = 177.5

Mode = 178

## Q4
Q4. Find the standard deviation for the given data:
[178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]

    To find the standard deviation for the given data [178, 177, 176, 177, 178.2, 178, 175, 179, 180, 175, 178.9, 176.2, 177, 172.5, 178, 176.5], you can follow these steps:

    1.Calculate the mean of the data. In this case, we found the mean to be approximately 176.68125 (as calculated in the previous question).

    2.For each data point, subtract the mean and square the result.

(178 - 176.68125)^2 = 1.3500440625

(177 - 176.68125)^2 = 0.1017965625

(176 - 176.68125)^2 = 0.4724600625

(177 - 176.68125 )^2 = 0.1017965625

(178.2 - 176.68125)^2 = 2.2955290625

(178 - 176.68125)^2 = 1.7339790625

(175 - 176.68125)^2 = 2.8366000625

(179 - 176.68125)^2 = 5.3200440625

(180 - 176.68125)^2 = 11.1063640625

(175 - 176.68125)^2 = 2.8366000625

(178.9 - 176.68125)^2 = 4.9244640625

(176.2 - 176.68125)^2 = 0.2311340625

(177 - 176.68125)^2 = 0.1017965625

(172.5 - 176.68125)^2 = 17.6215440625

(178 - 176.68125)^2 = 1.7339790625

(176.5 - 176.68125)^2 = 0.0330500625


    3.Sum up all the squared differences.

1.3500440625 + 0.1017965625 + 0.4724600625 + 0.1017965625 + 2.2955290625 + 1.7339790625 + 2.8366000625 + 5.3200440625 + 11.1063640625 + 2.8366000625 + 4.9244640625 + 0.2311340625 + 0.1017965625 + 17.6215440625 + 1.7339790625 + 0.0330500625 = 52.1563375

     4.Divide the sum by the total number of data points (16 in this case).

52.1563375 / 16 = 3.25977109375

     5.Take the square root of the result from step 4 to obtain the standard deviation.

√(3.25977109375) ≈ 1.806935

     Therefore, the standard deviation for the given data is approximately 1.806935.

## Q5
Q5. How are measures of dispersion such as range, variance, and standard deviation used to describe
the spread of a dataset? Provide an example.

    Measures of dispersion, such as range, variance, and standard deviation, are used to describe the spread or variability of a dataset. They provide information about how the data points are scattered or spread out around the measures of central tendency. Here's an example to illustrate their usage:

    Let's consider a dataset representing the scores of students in a math exam: [80, 85, 90, 75, 95]. We will calculate the range, variance, and standard deviation for this dataset.

    1.Range: The range is the simplest measure of dispersion and is calculated as the difference between the maximum and minimum values in the dataset. In this case, the maximum score is 95, and the minimum score is 75. Thus, the range is 95 - 75 = 20.

    2.Variance: The variance measures the average squared deviation of each data point from the mean. To calculate the variance, we follow these steps:

a) Calculate the mean of the dataset:
Mean = (80 + 85 + 90 + 75 + 95) / 5 = 85

b) Calculate the squared deviation for each data point:
(80 - 85)^2 = 25
(85 - 85)^2 = 0
(90 - 85)^2 = 25
(75 - 85)^2 = 100
(95 - 85)^2 = 100

c) Calculate the average of the squared deviations:
Variance = (25 + 0 + 25 + 100 + 100) / 5 = 50

     3.Standard Deviation: The standard deviation is the square root of the variance and provides a more interpretable measure of dispersion. In this case, the standard deviation is the square root of the variance calculated above:

 Standard Deviation = √(50) ≈ 7.07

     These measures of dispersion provide insights into the spread of the data. In our example, the range of 20 indicates that the scores vary by 20 points. The variance of 50 suggests that the scores deviate, on average, by approximately 7.07 points from the mean. The standard deviation of 7.07 provides a more intuitive understanding of the spread, indicating that the scores are dispersed around the mean by an average of 7.07 points.

## Q6
Q6. What is a Venn diagram?

     A Venn diagram is a visual representation of the relationships between different sets or groups of objects or concepts. It consists of overlapping circles or other shapes, each representing a set, and the overlapping regions representing the intersection of those sets.

     Venn diagrams are widely used in various fields, including mathematics, logic, statistics, and data analysis, to illustrate the logical relationships and overlaps between different categories or groups. They help in visualizing and understanding the relationships of elements within and between sets.

  The key elements of a Venn diagram include:

    1.Circles (or other shapes): Each circle represents a set or category. The elements belonging to that set are placed within the circle.

    2.Overlapping regions: The overlapping regions of the circles represent the intersection of the sets, i.e., elements that are common to multiple sets.

    3.Uncommon regions: The areas outside the overlapping regions represent elements that belong to individual sets but not to the other sets.

     Venn diagrams are often used to analyze and compare data, identify commonalities or differences between sets, solve logical problems, and aid in decision-making. They provide a visual tool to organize information and illustrate the relationships between various entities or concepts.

## Q7
Q7. For the two given sets A = (2,3,4,5,6,7) & B = (0,2,6,8,10). Find:
(i) A B
(ii) A ⋃ B

    To find the set operations A ∩ B (intersection) and A ∪ B (union) for the given sets:

    Set A = {2, 3, 4, 5, 6, 7}
    Set B = {0, 2, 6, 8, 10}

  (i) A ∩ B (Intersection):
  
    The intersection of two sets consists of elements that are common to both sets. In this case, the intersection of sets A and B is the set of elements that appear in both sets.

A ∩ B = {2, 6}

   (ii) A ∪ B (Union):

     The union of two sets consists of all unique elements from both sets, without repetition. In other words, it combines all the elements from both sets into a single set.

A ∪ B = {0, 2, 3, 4, 5, 6, 7, 8, 10}

    Therefore:
    
  (i) A ∩ B = {2, 6}
  
  (ii) A ∪ B = {0, 2, 3, 4, 5, 6, 7, 8, 10}

## Q8
Q8. What do you understand about skewness in data?

    Skewness is a statistical measure that describes the asymmetry or lack of symmetry in a distribution of data. It provides information about the shape of the distribution, specifically whether it is skewed to the left (negative skewness), skewed to the right (positive skewness), or symmetrical (zero skewness).

    Here's what you need to understand about skewness in data:

     1.Negative Skewness: Also known as left-skewed or left-tailed distribution, negative skewness occurs when the tail of the distribution extends towards the left, indicating that the majority of the data points are concentrated on the right side. In a negatively skewed distribution, the mean is typically less than the median. The left tail is longer and stretched out compared to the right tail.

      2.Positive Skewness: Also known as right-skewed or right-tailed distribution, positive skewness occurs when the tail of the distribution extends towards the right, indicating that the majority of the data points are concentrated on the left side. In a positively skewed distribution, the mean is typically greater than the median. The right tail is longer and stretched out compared to the left tail.

      3.Zero Skewness: Zero skewness indicates a symmetrical distribution where the data points are evenly distributed around the mean. In a symmetrical distribution, the mean, median, and mode are all equal. The tails on both sides of the distribution are of equal length.

       Skewness is a crucial measure as it provides insights into the shape and characteristics of the data distribution. It helps in understanding whether the data is concentrated towards one side or evenly spread. Skewed data can have implications on data analysis and modeling, as it can impact the accuracy of certain statistical assumptions or techniques. Hence, it is essential to consider skewness when interpreting and analyzing data.

## Q9
Q9. If a data is right skewed then what will be the position of median with respect to mean?

    If a data set is right-skewed (positively skewed), the median will typically be lower than the mean.

     In a right-skewed distribution, the tail of the data extends towards the higher values, indicating that there are relatively few extreme high values that pull the mean in that direction. As a result, the mean is typically higher than the median.

      To visualize this, imagine a right-skewed distribution where most of the data is concentrated on the left side, and a few extremely high values stretch the right tail. Since the median represents the middle value in the data set, it is less influenced by the extreme values in the right tail. On the other hand, the mean takes into account the magnitude of each value, including the extreme high values, causing it to be pulled towards the higher end of the distribution.

       Therefore, in a right-skewed distribution, the median will typically be lower than the mean, indicating that the center of the data is shifted towards the lower values.

## Q10
Q10. Explain the difference between covariance and correlation. How are these measures used in
statistical analysis?

    Covariance and correlation are both statistical measures used to describe the relationship between variables. While they are related concepts, there are important differences between them.

 Covariance:
 
    Covariance measures the extent to which two variables vary together. It quantifies the direction and magnitude of the linear relationship between two variables. A positive covariance indicates a positive relationship (both variables tend to increase or decrease together), while a negative covariance indicates a negative relationship (one variable tends to increase while the other decreases). However, covariance does not provide a standardized measure, making it difficult to interpret the strength of the relationship.

 Correlation:
 
     Correlation is a standardized measure that quantifies the strength and direction of the linear relationship between two variables. Unlike covariance, correlation ranges between -1 and +1, providing a more intuitive understanding of the relationship. A correlation of +1 indicates a perfect positive linear relationship, a correlation of -1 indicates a perfect negative linear relationship, and a correlation of 0 indicates no linear relationship. Correlation also accounts for the scales of the variables, making it easier to compare the strength of relationships across different datasets.

 Use in Statistical Analysis:
 
      Covariance and correlation are widely used in statistical analysis to understand the relationship between variables and to make inferences and predictions. Some key uses include:

 Relationship Assessment:

Covariance and correlation help assess the strength and direction of the relationship between two variables. This information is crucial for understanding dependencies and making predictions.

 Variable Selection: 

In regression analysis, covariance and correlation can be used to identify which variables have a strong relationship with the dependent variable, aiding in the selection of important predictor variables.

 Data Exploration:

Covariance and correlation are useful in exploring relationships in datasets. By examining the relationships between variables, patterns and associations can be uncovered, helping in data understanding and hypothesis generation.

 Quality of Models:

Covariance and correlation are used to assess the quality and validity of statistical models. Correlation coefficients close to 1 or -1 indicate a strong relationship, making the model more reliable.

    It is important to note that correlation does not imply causation. Even if two variables have a strong correlation, it does not necessarily mean that one variable causes the other. Careful interpretation and consideration of other factors are required to establish causal relationships.

## Q11
Q11. What is the formula for calculating the sample mean? Provide an example calculation for a
dataset.

    The formula for calculating the sample mean is as follows:

    Sample Mean = (Sum of all data points) / (Number of data points)

    To illustrate this, let's calculate the sample mean for the following dataset:

    Dataset: [12, 15, 18, 21, 24]

In [7]:
mean=(12 + 15 + 18 + 21 + 24)/5

In [8]:
print(mean)

18.0


Therefore, the sample mean for the given dataset is 18.

## Q12
Q12. For a normal distribution data what is the relationship between its measure of central tendency?

    In a normal distribution, the measure of central tendency, which includes the mean, median, and mode, is the same. All three measures are equal and coincide at the center of the distribution.

    This equality occurs because a normal distribution is symmetric, with the data points symmetrically distributed around the mean. The shape of a normal distribution is characterized by a bell-shaped curve, where the peak of the curve represents the mean, median, and mode. The mean is located at the center of the distribution, and due to the symmetry, the median and mode also align with the mean.

    To summarize:

     1.Mean: The mean of a normal distribution is equal to the median and mode. It represents the arithmetic average of all the data points.

     2.Median: The median of a normal distribution is equal to the mean and mode. It represents the middle value when the data points are arranged in ascending or descending order.

     3.Mode: The mode of a normal distribution is equal to the mean and median. It represents the most frequently occurring value in the dataset.

     Therefore, in a normal distribution, the measures of central tendency (mean, median, and mode) are all equal and located at the center of the distribution.

## Q13
Q13. How is covariance different from correlation?

    Covariance and correlation are both measures used to quantify the relationship between variables. While they are related concepts, there are important differences between them:

 ### Definition and Calculation:

    Covariance:  Covariance measures the extent to which two variables vary together. It is calculated by taking the average of the product of the differences between each variable's value and their respective means. The formula for covariance is: Cov(X, Y) = Σ[(X - X̄)(Y - Ȳ)] / (n - 1), where X and Y are variables, X̄ and Ȳ are their respective means, and n is the number of data points.

    Correlation: Correlation is a standardized measure that quantifies the strength and direction of the linear relationship between two variables. It is calculated by dividing the covariance by the product of the standard deviations of the two variables. The formula for correlation is: Corr(X, Y) = Cov(X, Y) / (σX * σY), where Cov(X, Y) is the covariance between X and Y, and σX and σY are the standard deviations of X and Y, respectively.

 ### Scale Independence:
     Covariance: Covariance is scale-dependent, meaning it is influenced by the units of measurement of the variables. Therefore, the magnitude of the covariance can vary depending on the scale of the variables.
     
     Correlation: Correlation is scale-independent and ranges between -1 and 1. It provides a standardized measure of the relationship between variables, making it easier to compare and interpret across different datasets.
 
 ### Interpretation:
    
    Covariance: Covariance values can be positive, negative, or zero. A positive covariance indicates a positive relationship (both variables tend to increase or decrease together), a negative covariance indicates a negative relationship (one variable tends to increase while the other decreases), and a covariance of zero indicates no linear relationship. However, it is challenging to interpret the strength of the relationship based on covariance alone.
     
     Correlation: Correlation coefficients range between -1 and 1. A correlation of +1 indicates a perfect positive linear relationship, a correlation of -1 indicates a perfect negative linear relationship, and a correlation of 0 indicates no linear relationship. The magnitude of the correlation coefficient indicates the strength of the relationship, with values closer to 1 or -1 indicating a stronger linear relationship.

     In summary, covariance measures the extent to which two variables vary together, while correlation provides a standardized measure of the strength and direction of the linear relationship between variables. Correlation is scale-independent and easier to interpret, making it a more widely used measure in statistical analysis.

## Q14
Q14. How do outliers affect measures of central tendency and dispersion? Provide an example.

    Outliers can have a significant impact on measures of central tendency and dispersion. They can distort the calculations and lead to misleading interpretations of the data. Here's how outliers affect these measures:

    Measures of Central Tendency:

    Mean: Outliers can substantially influence the mean since it takes into account the value of each data point. If there are extreme values in the dataset, the mean can be pulled towards those values, resulting in a skewed estimate of the central tendency.
    
    Median: The median is less affected by outliers since it only considers the middle value(s) in the dataset. Outliers have little impact on the median unless they are located within the middle values themselves.
   
    Mode: The mode is generally unaffected by outliers since it represents the most frequently occurring value(s) in the dataset. Outliers that occur rarely or as one-time extreme values do not significantly alter the mode.

 Measures of Dispersion:

    Range: Outliers can substantially increase the range since it is calculated as the difference between the maximum and minimum values in the dataset. If an outlier is extremely high or low, it can significantly increase the spread of the data.
    
    Variance: Outliers can increase the variance since it measures the average squared deviation of each data point from the mean. Outliers with large deviations from the mean contribute significantly to the variance.
    
    Standard Deviation: Like variance, outliers can increase the standard deviation since it is the square root of the variance. Outliers with large deviations from the mean can inflate the standard deviation.

 Example:
      
      Consider a dataset of exam scores: [85, 90, 92, 88, 95, 98, 85, 92, 94, 200]. In this dataset, the outlier value of 200 is significantly higher than the other scores. Let's see how outliers affect the measures of central tendency and dispersion:

    Mean: The outlier of 200 will pull the mean upwards, resulting in an inflated estimate of the average performance.

    Median: The median will be less affected by the outlier since it only considers the middle value, which remains unchanged.

    Mode: The mode will be unaffected by the outlier since it represents the most frequently occurring value, which remains the same.

    Range: The range will be significantly affected by the outlier since it increases the gap between the maximum value (200) and the minimum value (85).

    Variance: The variance will be influenced by the outlier as it contributes a large squared deviation from the mean.

    Standard Deviation: The standard deviation will be affected by the outlier since it is derived from the variance.

    In this example, the outlier has the potential to skew the mean, increase the range, variance, and standard deviation. However, the median and mode remain relatively unaffected by the presence of the outlier.